src/app/services/auth.service.ts

```diff
@@ -19,6 +19,17 @@ export class AuthService {
     return !!token;
   }
 
-  get identityClaims() { return this.oauthService.getIdentityClaims(); }
+  get identityClaims() { return this.oauthService.getIdentityClaims() || {}; }
 
+  roleMatch(allowedRoles: Array<String>): boolean {
+    let isMatch = false;
+    const userRoles = this.identityClaims['user_roles']  as Array<String> || [];
+    allowedRoles.forEach(role => {
+      if (userRoles.includes(role)) {
+        isMatch = true;
+        return;
```

Comment:
I don't think forEach will break out very cleanly with this method. We can make this a bit more concise using .some()
the first two answers here are a pretty good explanation
https://stackoverflow.com/questions/16312528/check-if-an-array-contains-any-element-of-another-array-in-javascript

Reply:
Will do.